lib/mysql_framework/connector.rb

```diff
@@ -3,19 +3,50 @@
 module MysqlFramework
   class Connector
     def initialize(options = {})
-      @connection_pool = ::Queue.new
-
       @options = default_options.merge(options)
 
       Mysql2::Client.default_query_options.merge!(symbolize_keys: true, cast_booleans: true)
     end
 
-    # This method is called to fetch a client from the connection pool or create a new client if no idle clients
-    # are available.
+    # This method is called to setup a pool of MySQL connections.
+    def setup
+      @connection_pool = ::Queue.new
+
+      start_pool_size.times { @connection_pool.push(Mysql2::Client.new(@options)) }
+
+      @created_connections = start_pool_size
+    end
+
+    # This method is called to close all MySQL connections in the pool and dispose of the pool itself.
+    def dispose
+      return if @connection_pool.nil?
+
+      until @connection_pool.empty?
+        conn = @connection_pool.pop(true)
+        conn.close
```

Comment:
I'm not sure how this works, but is there a need to add some sort of protection if the connection doesn't close? 
Something to force close it maybe?
Or put it back in the pool if it doesn't close so that it retries later?
(I might be talking nonsense)

Reply:
Maybe a rescue block to put it back into the pool i guess. However if for whatever reason the connection fails to close, would that not indicate there's something wrong with it so would we really want it back into the pool? However then we would be left with an open connection which isn't in the pool, correct?